### PR TITLE
Documenting Imports from Serato 

### PR DESCRIPTION
**Pull request by hannydevelop**
This is a draft pull request for https://github.com/mixxxdj/manual/issues/137 . I deleted the last branch i created and the PR because i ran into some messy complications.

The subheadings ```setup``` would describe steps on how the user can be able to import there serato library and metadata into Mixxx. Typos would be fixed but first, the purpose of this PR is to show the structure of the documentation I am currently writing. Please feel free to raise suggestions and changes that you want me to implement.

### CODE REVIEW

**Comment by hannydevelop** [comment]
Hello @ronso0 what kind of image do you think should be added to this piece for easy explanation? I see that most of the checks failed, would work on them immediately.

**Comment by hannydevelop** [comment]
Thank you @Holzhaus, the guide you gave was very useful!

File: source/chapters/import_from_serato.rst [hunk 1]
```diff
@@ -0,0 +1,83 @@
+.. _import-serato-metadata:
+
+Importing metadata from Serato
+*************
```
Comment (Holzhaus) [inline source/chapters/import_from_serato.rst:4]: Please have a look at the "netlify/mixxx-manual/deploy-preview" check below this PR. If you click on it, you can see the following warning:

```
9:53:50 PM: Warning, treated as error:
9:53:50 PM: /opt/build/repo/source/chapters/import_from_serato.rst:4:Title underline too short.
9:53:50 PM: Importing metadata from Serato
9:53:50 PM: *************
```

So the fix for that would be:

```suggestion
Importing metadata from Serato
******************************
```

**Review by Holzhaus — CHANGES_REQUESTED** [review]
Thank you. I think this should be integrated into the "Using external libraries" subheading in the "The Mixxx library" section: https://manual.mixxx.org/2.2/en/chapters/library.html#itunes-traktor-rhythmbox-banshee-using-external-libraries

**Comment by hannydevelop** [comment]
> Thank you. I think this should be integrated into the "Using external libraries" subheading in the "The Mixxx library" section: https://manual.mixxx.org/2.2/en/chapters/library.html#itunes-traktor-rhythmbox-banshee-using-external-libraries

ok! I would do that!

File: source/chapters/import_from_serato.rst (hunk 1 above)
Comment (hannydevelop) [reply source/chapters/import_from_serato.rst:4]: Thank you so much for the hint! i would amend the last commit just as @ronso0 taught me

**Comment by hannydevelop** [comment]
Changes implemented as requested! @jan 3 tests passed, 3 neutral

File: source/chapters/library.rst [hunk 2]
```diff
@@ -866,6 +866,90 @@ different location. Select :guilabel:`Use Default Library` to reset.
...
+.. _import-serato-metadata:
+
+Importing metadata from Serato
+------------------------------
+
+.. sectionauthor::
+   Ukpai Ugochi <outreachy applicant>
+
+Serato produces DJ softwares used for digital music mixing as well as digital vinyl record
+mixing for DVS records. While Serato is neither open source or customizable, It is possible
```
Comment (ywwg) [inline source/chapters/library.rst:878]: for ease of reviewing, can you put every sentence on its own line?

Comment (ywwg) [reply source/chapters/library.rst:878]: (without word wrap.  That way each line of the pull request is a single sentence)

Comment (hannydevelop) [reply source/chapters/library.rst:878]: Am on that!

**Comment by hannydevelop** [comment]
Hello 
Capture I have Mixxx 2.2.4 installed for windows and haven't checked on the steps i listed about importing Serato libraries from USB drive. It's late in the evening here since our tome zone is GMT+1. Would test the steps tomorrow and give feedback. Please review and give feedbacks, i would be so glad to work on the changes ASAP.

File: source/chapters/library.rst [hunk 3]
```diff
@@ -866,6 +867,77 @@ different location. Select :guilabel:`Use Default Library` to reset.
...
+
+Imports not currently supported
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Support for importing track overview waveform image stored in Serato into Mixxx has not been provided in Mixxx 2.3. 
+Serato flip macros cannot be imported either because flip does not offer that kind of functionality yet.
+
+Currently, you cannot write cues from Mixxx into Serato metadata tags. 
+We are working hard to implemment support for exporting Serato metadata tags in upcoming Mixxx versions, stay tunned.
+
```
Comment (Holzhaus) [inline source/chapters/library.rst:940]: Thank you for writing this. However, the phrasing is a bit "marketing-like" and for the manual we can make this a lot shorter and more formal.

I think we just need to describe the following things briefly:
- It's possible to read Serato libraries from your hard disk and USB drives, including crates. Smart crates are not supported at the moment.
- Mixxx will also read Serato metadata from file tags and will import the beatgrid and hotcues from the tags. Saved Loops will be imported too, but as of Mixxx 2.3 only the first loop is usable.
- If you want to reimport the metadata for files, you can right-click on a track and select Metadata > Import from file tag. If the track has Serato Metadata, all existing cue points will be overwritten.
- Exporting the beatgrid and hotcues to Serato is currently not supported.

**Review by Holzhaus — CHANGES_REQUESTED** [review]

File: source/chapters/library.rst [hunk 4]
```diff
@@ -840,6 +840,7 @@ iTunes, Traktor, Rhythmbox, Banshee - Using external libraries
 
 .. sectionauthor::
    S.Brandt <s.brandt@mixxx.org>
+   Ukpai Ugochi <outreachy applicant>
 
 Supported libraries:
```
[the remark this answers is not in the document]
Comment (hannydevelop) [reply source/chapters/library.rst:845]: Ok! I will do that now!

File: source/chapters/library.rst (hunk 3 above)
Comment (hannydevelop) [reply source/chapters/library.rst:940]: ok! I would look into this now!

File: source/chapters/library.rst (hunk 4 above)
[the remark this answers is not in the document]
Comment (hannydevelop) [reply source/chapters/library.rst:845]: > Serato should be listed here too.
> 
> Also, the title of the section should probably be changed, because if we also add Serato and Rekordbox it becomes too long. How about "Using External Libraries" or "Importing libraries from other software" or something like that?

I have done this! I added Serato to the title, I didn't add RekordBok because nothing was talked about it in that section. The issue for this pull request is **Documenting Imports from Serato**, nothing was said about **RekordBox**. The title of this section already has **Using external libraries** in it. Please let me know if there's any title you think would fit this section more, i would be glad to add it.

File: source/chapters/library.rst (hunk 3 above)
Comment (hannydevelop) [reply source/chapters/library.rst]: > Thank you for writing this. However, the phrasing is a bit "marketing-like" and for the manual we can make this a lot shorter and more formal.
> 
> I think we just need to describe the following things briefly:
> 
> * It's possible to read Serato libraries from your hard disk and USB drives, including crates. Smart crates are not supported at the moment.
> * Mixxx will also read Serato metadata from file tags and will import the beatgrid and hotcues from the tags. Saved Loops will be imported too, but as of Mixxx 2.3 only the first loop is usable.
> * If you want to reimport the metadata for files, you can right-click on a track and select Metadata > Import from file tag. If the track has Serato Metadata, all existing cue points will be overwritten.
> * Exporting the beatgrid and hotcues to Serato is currently not supported.

I just worked on this, thank you for your time and the **honest feedback you gave**. Changes have been done to effect this, please let me know if there's anything to add or remove to make this documentation better.

Don't you think we need an image for this section? Any suggestions?

File: source/chapters/library.rst (hunk 4 above)
[the remark this answers is not in the document]
Comment (Holzhaus) [reply source/chapters/library.rst:845]: Thanks. What I meant is that we also support Rekordbox, and if we are going to add that to the section title too it would become a bit long. That's why I suggested to removed all application names from the section and just name it something like "Using libraries from other software".

Also, there is a list "Supported libraries" and Serato is currently missing from that list. Could you add that please? A logo can be found here: https://github.com/mixxxdj/mixxx/blob/master/res/images/library/ic_library_serato.svg

File: source/chapters/library.rst [hunk 5]
```diff
@@ -866,6 +867,86 @@ different location. Select :guilabel:`Use Default Library` to reset.
...
+
+From Mixxx 2.3, tracks in your Serato library and crates can now be imported into Mixxx library table.
+Tracks can also be loaded into deck without needing to add the music directory in the preferences. 
+You can also import your Serato library into Mixxx that are in USB drives for a portable Serato library.
+
+**How to import Serato libraries from external USB drive**
+
+* Copy music library from Serato into external USB drive
+* Plug in USB drive while Mixxx is running
+* Click on Serato in the library, all records on USB from Serato will be visible (attach as many as you like!)
```
Comment (Holzhaus) [inline source/chapters/library.rst:883]:
```suggestion
* Click on Serato in the library. All USB drives containing a Serato library will be detected automatically.
```

File: source/chapters/library.rst [hunk 6]
```diff
@@ -866,6 +867,86 @@ different location. Select :guilabel:`Use Default Library` to reset.
...
+* Then click on the desired removable device, and all your folders and playlists from Serato will be revealed for you to play from
```
Comment (Holzhaus) [inline source/chapters/library.rst:884]:
```suggestion
*  For each removable device, the library and all crates from Serato will accessible.
```

File: source/chapters/library.rst [hunk 7]
```diff
@@ -866,6 +867,86 @@ different location. Select :guilabel:`Use Default Library` to reset.
   Mac partition mounted in Linux, you can load it and use your iTunes tracks and
   playlists as well.
 
+.. _import-serato-metadata:
+
+Importing metadata from Serato
+------------------------------
```
Comment (Holzhaus) [inline source/chapters/library.rst:873]: The section more than just metadata tags.

What do think about this?

```suggestion
.. _library-serato:

Using the Serato library
------------------------
```

File: source/chapters/library.rst [hunk 8]
```diff
@@ -866,6 +867,86 @@ different location. Select :guilabel:`Use Default Library` to reset.
...
+
+From Mixxx 2.3, tracks in your Serato library and crates can now be imported into Mixxx library table.
+Tracks can also be loaded into deck without needing to add the music directory in the preferences. 
+You can also import your Serato library into Mixxx that are in USB drives for a portable Serato library.
```
Comment (Holzhaus) [inline source/chapters/library.rst:877]:
```suggestion
Mixxx supports reading your Serato library and crates both from your local hard drive as well as portable USB drives. Smart crates are currently not supported.

Serato stores information such as hot cues, track color and beatgrid in the file tags.
Regardless of whether a track is loaded into deck from the libray or directly from the file browser, Mixxx will import that information automatically if present.

Saved loops are imported as well, but since Mixxx is lacking support for saved loops, only the first loop is usable in Mixxx.
All other saved loops can be used like regular hotcues for now.
   
Waveforms, Gain values and "Flips" are not imported from Serato.
   
.. note:: Due to differences in the metadata format, importing Serato information from Ogg Vorbis files is currently not supported.
```

File: source/chapters/library.rst [hunk 9]
```diff
@@ -866,6 +867,86 @@ different location. Select :guilabel:`Use Default Library` to reset.
...
+* Then click on the desired removable device, and all your folders and playlists from Serato will be revealed for you to play from
+
+Import your Beatgrid and Hotcues
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Track cue points are saved in special file tags as well as the track metadata. 
+You can now import your track cue points and all of it's metadata from Serato to Mixxx.
+
+Once you import your Serato library into Mixxx and load a track for the first time, the positions, labels and colors of your hotcues, other metadatas and beatgrid are automatically imported.
+
```
Comment (Holzhaus) [inline source/chapters/library.rst:893]: I think we can keep this shorter. Everything explained here has already been mentioned above.

```suggestion
```

File: source/chapters/library.rst [hunk 10]
```diff
@@ -866,6 +867,86 @@ different location. Select :guilabel:`Use Default Library` to reset.
...
+This action will clear your existing cuepoints in Mixxx if the track has any Serato hotcues.
+
+**To reimport metadata for files**
+
+* Right click on the track and select metadata
+* Click on Import from file tag. If the track has Serato Metadata, all existing cue points will be overwritten
+
+Decodes may slightly detect different track start and end times for files from different sources, this could possibly be because of the presence of countless encoders and decoders for MP3 and M4A/AAC.
+This problem may cause your cues to be shifted up by a few milliseconds. 
+While we are working very hard to mitigate this problem, we have added a way to shift all cues for a track at once.
```
Comment (Holzhaus) [inline source/chapters/library.rst:904]:
```suggestion
Mixxx might not always be able to mitigate this problem, but it's possible shift all cues for a track at once as a workaround. 
```

File: source/chapters/library.rst [hunk 11]
```diff
@@ -866,6 +867,86 @@ different location. Select :guilabel:`Use Default Library` to reset.
...
+
+If you have already added a track to your Mixxx library before support for reading Serato's hotcues was added, you can reimport metadata from Serato via the track context menu.
+This action will clear your existing cuepoints in Mixxx if the track has any Serato hotcues.
+
+**To reimport metadata for files**
+
+* Right click on the track and select metadata
+* Click on Import from file tag. If the track has Serato Metadata, all existing cue points will be overwritten
+
+Decodes may slightly detect different track start and end times for files from different sources, this could possibly be because of the presence of countless encoders and decoders for MP3 and M4A/AAC.
```
Comment (Holzhaus) [inline source/chapters/library.rst:902]:
```suggestion
Decoders may slightly detect different track start and end times for files from different sources, this could possibly be because of the presence of countless encoders and decoders for MP3 and M4A/AAC.
```

File: source/chapters/library.rst [hunk 12]
```diff
@@ -866,6 +867,86 @@ different location. Select :guilabel:`Use Default Library` to reset.
...
+With this in place, you can start using your loops once Mixxx 2.4 is out and avoid reimporting track meta data.
+
+Imports not currently supported
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Support for importing track overview waveform image stored in Serato into Mixxx and smart crates has not been provided in Mixxx 2.3. 
+Serato flip macros cannot be imported either because Mixxx does not offer that kind of functionality yet.
+
+Currently, you cannot write cues from Mixxx into Serato metadata tags. 
+We are working hard to implemment support for exporting Serato metadata tags in upcoming Mixxx versions, stay tunned.
```
Comment (Holzhaus) [inline source/chapters/library.rst:948]: With the changes I suggested above this can be removed.

```suggestion
```

**Review by Holzhaus — CHANGES_REQUESTED** [review]
Thank you. I think we should shorten it a bit to make it easier to find the important information.

File: source/chapters/library.rst (hunk 12 above)
Comment (hannydevelop) [reply source/chapters/library.rst:948]: Thank you so much @Holzhaus. I would implement this as soon as possible

File: source/chapters/library.rst (hunk 4 above)
[the remark this answers is not in the document]
Comment (hannydevelop) [reply source/chapters/library.rst:845]: > Thanks. What I meant is that we also support Rekordbox, and if we are going to add that to the section title too it would become a bit long. That's why I suggested to removed all application names from the section and just name it something like "Using libraries from other software".
> 
> Also, there is a list "Supported libraries" and Serato is currently missing from that list. Could you add that please? A logo can be found here: https://github.com/mixxxdj/mixxx/blob/master/res/images/library/ic_library_serato.svg

Sure, I would change the title of the section (it certainly does look long). I would also add Serato to the list of supported libraries in Mixxx manual-2.3x. ```"Using libraries from other software"``` the title should fit in, especially if Mixxx supports other software then there would be no need to change title again.

File: source/chapters/library.rst (hunk 5 above)
Comment (hannydevelop) [reply source/chapters/library.rst:883]: I would work on all the changes you requested immediately and notify you when am done. Thank you so much for your help!

File: source/chapters/library.rst (hunk 7 above)
Comment (hannydevelop) [reply source/chapters/library.rst:873]: Sure! Since the section talks of more than just metadata tags

File: source/chapters/library.rst (hunk 12 above)
Comment (hannydevelop) [reply source/chapters/library.rst:948]: > With the changes I suggested above this can be removed.

Ok! I would remove the table and effect other changes you requested for

File: source/chapters/library.rst (hunk 7 above)
Comment (Holzhaus) [reply source/chapters/library.rst:873]: By the way, you can apply apply these suggestions directly from GitHub web interface: https://docs.github.com/en/free-pro-team@latest/github/collaborating-with-issues-and-pull-requests/incorporating-feedback-in-your-pull-request#applying-suggested-changes

Comment (hannydevelop) [reply source/chapters/library.rst:873]: > By the way, you can apply apply these suggestions directly from GitHub web interface: https://docs.github.com/en/free-pro-team@latest/github/collaborating-with-issues-and-pull-requests/incorporating-feedback-in-your-pull-request#applying-suggested-changes

Thank you so much! I would read this up and implement in future suggestions. Thank you so much!

File: source/chapters/library.rst (hunk 10 above)
Comment (hannydevelop) [reply source/chapters/library.rst:904]: ```Mixxx might not always be able to mitigate this problem, but it's possible **to** shift all cues for a track at once as a workaround```.

You missed **to** in this sentence, i added it to the commit

File: source/chapters/library.rst (hunk 12 above)
Comment (hannydevelop) [reply source/chapters/library.rst]: > With the changes I suggested above this can be removed.

Hello @Holzhaus I just implemented the changes, Please review!

File: source/chapters/library.rst (hunk 4 above)
[the remark this answers is not in the document]
Comment (Holzhaus) [reply source/chapters/library.rst:845]: > Also, there is a list "Supported libraries" and Serato is currently missing from that list. Could you add that please? A logo can be found here: https://github.com/mixxxdj/mixxx/blob/master/res/images/library/ic_library_serato.svg

Ping.

File: source/chapters/library.rst [hunk 13]
```diff
@@ -866,6 +867,48 @@ different location. Select :guilabel:`Use Default Library` to reset.
...
+
+* Right click on the track and select metadata
+* Click on Import from file tag. If the track has Serato Metadata, all existing cue points will be overwritten
+
+Decoders may slightly detect different track start and end times for files from different sources, this could possibly be because of the presence of countless encoders and decoders for MP3 and M4A/AAC.
+This problem may cause your cues to be shifted up by a few milliseconds. 
+
+Mixxx might not always be able to mitigate this problem, but it's possible to shift all cues for a track at once as a workaround.
+
 .. seealso:: External libraries can be disabled under
```
Comment (Holzhaus) [inline source/chapters/library.rst:912]: I think this is unrelated to the Serato library import and should be moved up, right below the "Playing a track from an external library will add it to your Mixxx library" note.

**Review by Holzhaus — CHANGES_REQUESTED** [review]
Thanks!

File: source/chapters/library.rst (hunk 13 above)
Comment (hannydevelop) [reply source/chapters/library.rst:912]: > I think this is unrelated to the Serato library import and should be moved up, right below the "Playing a track from an external library will add it to your Mixxx library" note.

Hello @Holzhaus changes have been implemented, please review! Thank you

File: source/chapters/library.rst (hunk 4 above)
[the remark this answers is not in the document]
Comment (hannydevelop) [reply source/chapters/library.rst:845]: > > Also, there is a list "Supported libraries" and Serato is currently missing from that list. Could you add that please? A logo can be found here: https://github.com/mixxxdj/mixxx/blob/master/res/images/library/ic_library_serato.svg
> 
> Ping.

I am working on this now!

[the remark this answers is not in the document]
Comment (hannydevelop) [reply source/chapters/library.rst:845]: > > > Also, there is a list "Supported libraries" and Serato is currently missing from that list. Could you add that please? A logo can be found here: https://github.com/mixxxdj/mixxx/blob/master/res/images/library/ic_library_serato.svg
> > 
> > 
> > Ping.
> 
> I am working on this now!

Hello @Holzhaus I have done this! Please review.

File: source/chapters/library.rst (hunk 13 above)
Comment (Holzhaus) [reply source/chapters/library.rst:913]: Sorry if this was unclear, but I meant the "External libraries can be disabled" note, not the stuff about encoders.

Comment (hannydevelop) [reply source/chapters/library.rst]: > Sorry if this was unclear, but I meant the "External libraries can be disabled" note, not the stuff about encoders.

No problems, I just effected the change, please review

**Comment by hannydevelop** [comment]
@Holzhaus , @ronso0 can we merge this PR if there are no other reviews?

**Comment by ronso0** [comment]
@holzhaus I resign from thi sPR as I can't take care of this right now, also I you're more familiar with the interface

File: source/chapters/library.rst [hunk 14]
```diff
@@ -866,8 +871,47 @@ different location. Select :guilabel:`Use Default Library` to reset.
...
+
+* Copy music library from Serato into external USB drive
+* Plug in USB drive while Mixxx is running
+* Click on Serato in the library. All USB drives containing a Serato library will be detected automatically.
+* For each removable device, the library and all crates from Serato will accessible.
+
+Import your Beatgrid and Hotcues
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Serato libraries imported into Mixxx would have beatgrid, hotcues and other metadata imported immediately you load the track for the first time.
```
Comment (Holzhaus) [inline source/chapters/library.rst:901]: I think this is redundant can be removed without losing information.

```suggestion
```

File: source/chapters/library.rst [hunk 15]
```diff
@@ -866,8 +871,47 @@ different location. Select :guilabel:`Use Default Library` to reset.
...
+
+**To reimport metadata for files**
+
+* Right click on the track and select metadata
+* Click on Import from file tag. If the track has Serato Metadata, all existing cue points will be overwritten
+
+Decoders may slightly detect different track start and end times for files from different sources, this could possibly be because of the presence of countless encoders and decoders for MP3 and M4A/AAC.
+This problem may cause your cues to be shifted up by a few milliseconds. 
+
+Mixxx might not always be able to mitigate this problem, but it's possible to shift all cues for a track at once as a workaround.
```
Comment (Holzhaus) [inline source/chapters/library.rst:914]: This could be a `hint` to make it stand out more, but no idea if this is better. You decide.

File: source/chapters/library.rst [hunk 16]
```diff
@@ -866,8 +871,47 @@ different location. Select :guilabel:`Use Default Library` to reset.
...
+   
+Waveforms, Gain values and "Flips" are not imported from Serato.
+
+.. note:: Due to differences in the metadata format, importing Serato information from Ogg Vorbis files is currently not supported.
+
+**How to import Serato libraries from external USB drive**
+
+* Copy music library from Serato into external USB drive
+* Plug in USB drive while Mixxx is running
+* Click on Serato in the library. All USB drives containing a Serato library will be detected automatically.
```
Comment (Holzhaus) [inline source/chapters/library.rst:895]:
```suggestion
* Click on :guilabel:`Serato` in the library. All USB drives containing a Serato library will be detected automatically.
```

File: source/chapters/library.rst [hunk 17]
```diff
@@ -866,8 +871,47 @@ different location. Select :guilabel:`Use Default Library` to reset.
...
+
+Serato libraries imported into Mixxx would have beatgrid, hotcues and other metadata imported immediately you load the track for the first time.
+
+If you have already added a track to your Mixxx library before support for reading Serato's hotcues was added, you can reimport metadata from Serato via the track context menu.
+This action will clear your existing cuepoints in Mixxx if the track has any Serato hotcues.
+
+**To reimport metadata for files**
+
+* Right click on the track and select metadata
+* Click on Import from file tag. If the track has Serato Metadata, all existing cue points will be overwritten
```
Comment (Holzhaus) [inline source/chapters/library.rst:909]:
```suggestion
* Right click on the track to show the context menu
* Select :menuselection:`Metadata --> Import From File Tags`. If the track has Serato Metadata, all existing cue points will be overwritten
```

**Review by Holzhaus — CHANGES_REQUESTED** [review]

File: source/chapters/library.rst (hunk 17 above)
Comment (hannydevelop) [reply source/chapters/library.rst]: > 

changes have been implemented

File: source/chapters/library.rst (hunk 16 above)
Comment (hannydevelop) [reply source/chapters/library.rst]: changes have been implemented

File: source/chapters/library.rst (hunk 14 above)
Comment (hannydevelop) [reply source/chapters/library.rst]: changes have been implemented

File: source/chapters/library.rst (hunk 15 above)
Comment (hannydevelop) [reply source/chapters/library.rst]: I made it a ```hint```

**Comment by hannydevelop** [comment]
Hello @Holzhaus changes have been implemented, please review!

**Review by Holzhaus — CHANGES_REQUESTED** [review]
Thank you, almost done.

File: source/chapters/library.rst [hunk 18]
```diff
@@ -866,8 +871,41 @@ different location. Select :guilabel:`Use Default Library` to reset.
...
+* Click on :guilabel:`Serato` in the library. All USB drives containing a Serato library will be detected automatically.
+* For each removable device, the library and all crates from Serato will accessible.
+
+If you have already added a track to your Mixxx library before support for reading Serato's hotcues was added, you can reimport metadata from Serato via the track context menu.
+This action will clear your existing cuepoints in Mixxx if the track has any Serato hotcues.
+
+**To reimport metadata for files**
+
+* Right click on the track to show the context menu
+* Select :menuselection:`Metadata --> Import From File Tags`. If the track has Serato Metadata, all existing cue points will be overwritten
```
[the remark this answers is not in the document]
Comment (hannydevelop) [reply source/chapters/library.rst:904]: Done!

File: source/chapters/library.rst [hunk 19]
```diff
@@ -866,8 +871,42 @@ different location. Select :guilabel:`Use Default Library` to reset.
...
+Regardless of whether a track is loaded into deck from the libray or directly from the file browser, Mixxx will import that information automatically if present.
+
+Saved loops are imported as well, but since Mixxx is lacking support for saved loops, only the first loop is usable in Mixxx.
+All other saved loops can be used like regular hotcues for now.
+   
+Waveforms, Gain values and "Flips" are not imported from Serato.
+
+.. note:: Due to differences in the metadata format, importing Serato information from Ogg Vorbis files is currently not supported.
+
+**How to import Serato libraries from external USB drive**
```
Comment (Holzhaus) [inline source/chapters/library.rst:891]: The other instruction lists in this chapter don't use a bold font, so maybe this would fit a bit better:

```suggestion
To import Serato libraries from external USB drives:
```

File: source/chapters/library.rst [hunk 20]
```diff
@@ -866,8 +871,42 @@ different location. Select :guilabel:`Use Default Library` to reset.
...
+* Copy music library from Serato into external USB drive
+* Plug in USB drive while Mixxx is running
+* Click on :guilabel:`Serato` in the library. All USB drives containing a Serato library will be detected automatically.
+* For each removable device, the library and all crates from Serato will accessible.
+
+If you have already added a track containing Serato Metadata to your Mixxx library in a Mixxx version that didn't support reading that data, you can reimport it from the file's tags via the track context menu.
+
+.. warning:: This action will clear your existing cuepoints in Mixxx if the file contains Serato Metadata
+
+**To reimport metadata for files**
```
Comment (Holzhaus) [inline source/chapters/library.rst:902]:
```suggestion
To reimport metadata for files:
```

**Review by Holzhaus — APPROVED** [review]
LGTM, thank you.

@ywwg You assigned this PR to yourself, please have review and merge.

File: source/chapters/library.rst [hunk 21]
```diff
@@ -835,18 +835,20 @@ play. See :ref:`getting-started-analyze-library` for details.
...
 .. _library-third-party:
 
-iTunes, Traktor, Rhythmbox, Banshee - Using external libraries
-==============================================================
+Using libraries from other software
+===================================
 
 .. sectionauthor::
    S.Brandt <s.brandt@mixxx.org>
+   Ukpai Ugochi <outreachy applicant>
```
Comment (ywwg) [inline source/chapters/library.rst:843]: You can put your email address here if you want.

File: source/chapters/library.rst [hunk 22]
```diff
@@ -835,18 +835,20 @@ play. See :ref:`getting-started-analyze-library` for details.
...
 
 Supported libraries:
 
 * |ic_lib_itunes| `iTunes <https://www.apple.com/itunes/>`_ (Windows, macOS)
 * |ic_lib_traktor| `Traktor <https://www.native-instruments.com/en/catalog/traktor/>`_ (Windows, macOS)
 * |ic_lib_rhythmbox| `Rhythmbox <https://wiki.gnome.org/Apps/Rhythmbox>`_ (GNU/Linux)
 * |ic_lib_banshee| `Banshee <http://banshee.fm/>`_ (Windows, macOS, GNU/Linux)
+* |ic_lib_serato| `Serato <https://serato.com/>`_ (Windows, macOS, GNU/Linux)
```
Comment (ywwg) [inline source/chapters/library.rst:851]: serato is not supported on Linux

File: source/chapters/library.rst (hunk 19 above)
Comment (ywwg) [reply source/chapters/library.rst:891]: +1 to this

File: source/chapters/library.rst [hunk 23]
```diff
@@ -866,8 +871,42 @@ different location. Select :guilabel:`Use Default Library` to reset.
...
+Saved loops are imported as well, but since Mixxx is lacking support for saved loops, only the first loop is usable in Mixxx.
+All other saved loops can be used like regular hotcues for now.
+   
+Waveforms, Gain values and "Flips" are not imported from Serato.
+
+.. note:: Due to differences in the metadata format, importing Serato information from Ogg Vorbis files is currently not supported.
+
+**How to import Serato libraries from external USB drive**
+
+* Copy music library from Serato into external USB drive
```
Comment (ywwg) [inline source/chapters/library.rst:893]: since this is an ordered list of steps, we should use numbers instead of bullet points.

```
  1.  Copy music...
  1.  Plug in USB drive....
```

markdown will set the numbers correctly, you can make all of them "1." and it will work right.

File: source/chapters/library.rst (hunk 20 above)
Comment (ywwg) [reply source/chapters/library.rst:902]: +1

**Review by ywwg — CHANGES_REQUESTED** [review]
This looks really good! I left a few small comments.

File: source/chapters/library.rst (hunk 23 above)
Comment (Holzhaus) [reply source/chapters/library.rst:893]: This is RST, not Markdown. Just number them correctly :D

**Comment by hannydevelop** [comment]
Hello @ronso0, @Holzhaus, @ywwg My system got stolen that's why i haven't been able to make the changes.  I got a new one and don't know how to effect these changes now, any suggestions? Thank you!

File: source/chapters/library.rst (hunk 23 above)
Comment (hannydevelop) [reply source/chapters/library.rst:893]: Hello @ywwg can you request for a change with the numbered list so i commit them directly here? I lost my system (Stolen) I don't have access to the changes I have made already.

**Comment by Holzhaus** [comment]
You can install git and clone your fork of the manual repo again.

**Comment by hannydevelop** [comment]
> You can install git and clone your fork of the manual repo, the you can continue working on the manual from your computer.

Thank you @Holzhaus, would the forked copy contain changes i pushed that havn't been merged? I would do that and give feedback, thanks! thank you too @Be-ing!

**Comment by Holzhaus** [comment]
All changes you *pushed* to your fork are saved. Changes in your local worktree that you didn't commit or push yet are lost unfortunately.

File: source/chapters/library.rst (hunk 23 above)
Comment (hannydevelop) [reply source/chapters/library.rst]: done!

File: source/chapters/library.rst (hunk 22 above)
Comment (hannydevelop) [reply source/chapters/library.rst]: done!

File: source/chapters/library.rst (hunk 21 above)
Comment (hannydevelop) [reply source/chapters/library.rst]: done!

**Comment by hannydevelop** [comment]
> All changes you _pushed_ to your fork are saved. Changes in your local worktree that you didn't commit or push yet are lost unfortunately.

Thank you @Holzhaus all the changes i pushed to my fork was available. gladly i always push/commit immediately i effect a change so no changes were lost. Thank you so much for the hint, never knew forked copies retained changes that were not merged, am really glad.

**Comment by hannydevelop** [comment]
Hello @ywwg, @Holzhaus, @Be-ing I have effected the changes, what do you think?

File: source/chapters/library.rst (hunk 19 above)
Comment (Holzhaus) [reply source/chapters/library.rst:891]: Looks like you marked this as resolved without changing it?

File: source/chapters/library.rst (hunk 20 above)
Comment (Holzhaus) [reply source/chapters/library.rst]: This also marked as resolved but not fixed.

File: source/chapters/library.rst (hunk 19 above)
Comment (hannydevelop) [reply source/chapters/library.rst:891]: > Looks like you marked this as resolved without changing it?

Fixed now!

File: source/chapters/library.rst (hunk 20 above)
Comment (hannydevelop) [reply source/chapters/library.rst]: > This also marked as resolved but not fixed.

Fixed now, sorry.

**Comment by hannydevelop** [comment]
Hello @Holzhaus, kindly review.

Thank you!

File: source/chapters/library.rst [hunk 24]
```diff
@@ -866,8 +871,42 @@ different location. Select :guilabel:`Use Default Library` to reset.
...
+
+Using the Serato library
+------------------------
+
+Mixxx supports reading your Serato library and crates both from your local hard drive as well as portable USB drives. Smart crates are currently not supported.
+
+Serato stores information such as hot cues, track color and beatgrid in the file tags.
+Regardless of whether a track is loaded into deck from the libray or directly from the file browser, Mixxx will import that information automatically if present.
+
+Saved loops are imported as well, but since Mixxx is lacking support for saved loops, only the first loop is usable in Mixxx.
```
Comment (ywwg) [inline source/chapters/library.rst:884]: Maybe:

Saved loops are imported as well, but since Mixxx is lacking support for mulitple saved loops, only the first loop is usable in Mixxx.

(add the word "multiple")

**Review by ywwg — APPROVED** [review]
LGTM! just one more small change from me

**Review by Holzhaus — APPROVED** [review]
LGTM, thank you very much.

File: source/chapters/library.rst (hunk 24 above)
Comment (hannydevelop) [reply source/chapters/library.rst]: Changes effected @ywwg! Thank you for the reviews @Be-ing, @Holzhaus, @ronso0 and @ywwg.

**Review by Holzhaus — APPROVED** [review]

File: source/chapters/library.rst [hunk 25]
```diff
@@ -866,8 +871,42 @@ different location. Select :guilabel:`Use Default Library` to reset.
...
+
+Using the Serato library
+------------------------
+
+Mixxx supports reading your Serato library and crates both from your local hard drive as well as portable USB drives. Smart crates are currently not supported.
+
+Serato stores information such as hot cues, track color and beatgrid in the file tags.
+Regardless of whether a track is loaded into deck from the libray or directly from the file browser, Mixxx will import that information automatically if present.
+
+Saved loops are imported as well, but since Mixxx is lacking support for mulitple saved loops, only the first loop is usable in Mixxx.
```
Comment (Holzhaus) [inline source/chapters/library.rst:884]:
```suggestion
Saved loops are imported as well, but since Mixxx is lacking support for multiple saved loops, only the first loop is usable in Mixxx.
```

**Review by Holzhaus — CHANGES_REQUESTED** [review]
Just a typo.

**Review by Holzhaus — APPROVED** [review]

File: source/chapters/library.rst (hunk 25 above)
Comment (hannydevelop) [reply source/chapters/library.rst]: Done!
